### PR TITLE
Update liboqs readme to point to oqs-provider instead of deprecated openssl1.1.1 fork [skip ci]

### PR DESCRIPTION
**Pull request by Martyrshot**
Update liboqs readme to point to oqs-provider instead of deprecated openssl1.1.1 fork [skip ci]

This PR updates the OpenSSL link to point to oqs-provider now that OpenSSL1.1.1 is deprecated.

### CODE REVIEW

**Review by baentsch — APPROVED** [review]
Maybe wortwhile also removing the pointer to a project no longer actively maintained (openssh) and people not exactly relevant any more (Michele Mosca) pointing to PQSA and/or TSC instead?

**Comment by baentsch** [comment]
(side note: use '[skip ci]' to skip CI; 'skip-ci' doesn't)

**Comment by Martyrshot** [comment]
Ah, thanks for the tip. :-)

**Comment by Martyrshot** [comment]
> Maybe wortwhile also removing the pointer to a project no longer actively maintained (openssh) and people not exactly relevant any more (Michele Mosca) pointing to PQSA and/or TSC instead?

I think this would make a lot of sense. I would probably hold off from removing openssh until the list of projects we will continue supporting/won't continue supporting is finalized.

It wasn't clear to me how involved Michele was behind the scenes, but seeing as I started using/contributing to liboqs ~3 years ago and have never talked to them, I think it's as good a time as any haha. But there might be other reasons they is still listed, so I would defer to @dstebila on that one.

**Review by dstebila — APPROVED** [review]

**Comment by dstebila** [comment]
> > Maybe wortwhile also removing the pointer to a project no longer actively maintained (openssh) and people not exactly relevant any more (Michele Mosca) pointing to PQSA and/or TSC instead?
> 
> I think this would make a lot of sense. I would probably hold off from removing openssh until the list of projects we will continue supporting/won't continue supporting is finalized.

I think it's okay to remove the prominent link to OpenSSH, and I've replaced it with a link to our more general demos project.

> It wasn't clear to me how involved Michele was behind the scenes, but seeing as I started using/contributing to liboqs ~3 years ago and have never talked to them, I think it's as good a time as any haha. But there might be other reasons they is still listed, so I would defer to @dstebila on that one.

This is a good time to revisit that wording, as well as also pop in wording about the Linux Foundation, which I've now done as another commit on this branch.

File: README.md
Comment (baentsch) [inline README.md:33]: warning: broken link (-> .../oqs-demos).

File: README.md
Comment (baentsch) [inline README.md:35]: Hmm. Is "supported" the right term (no new contributors so far, [no feedback](https://github.com/orgs/open-quantum-safe/discussions/1689), [no promised input](https://github.com/open-quantum-safe/liboqs/pull/1650#discussion_r1462260229), ...)? What about "part of"? Yes, I know, PQCA is all-new, but 1 year "in the making...".

File: README.md
Comment (baentsch) [reply README.md:33]: @Martyrshot : Would you want to update the (link target) bug above before merge?

Comment (Martyrshot) [reply README.md]: Sorry about that @baentsch. Should be fixed in 2ecd18c

**Pull request merged**
